app/src/main/java/org/simple/clinic/summary/teleconsultation/messagebuilder/LongTeleconsultMessageBuilder.kt

```diff
@@ -0,0 +1,145 @@
+        .appendLine("")
+  }
+
+  private fun addLogTeleconsultSectionToMessage(patientTeleconsultationInfoLong: PatientTeleconsultationInfoLong, message: StringBuilder) {
+    message
+        .appendLine("3️⃣ *Log teleconsult*: https://app.simple.org/consult?r=${patientTeleconsultationInfoLong.teleconsultationId}&p=${patientTeleconsultationInfoLong.patientUuid}")
+        .appendLine("")
+  }
+
+  private fun addCallForTeleconsultSectionToMessage(
+      patientTeleconsultationInfoLong: PatientTeleconsultationInfoLong,
+      message: StringBuilder
+  ) {
+    if (patientTeleconsultationInfoLong.nursePhoneNumber.isNullOrBlank().not()) {
+      message
+          .appendLine("2️⃣ *Call for teleconsult*: ${patientTeleconsultationInfoLong.nursePhoneNumber}")
+          .appendLine("")
+    }
+  }
+
+  private fun addPrescriptionsSectionToMessage(
+      patientTeleconsultationInfo: PatientTeleconsultationInfoLong,
+      message: StringBuilder
+  ) {
+    if (patientTeleconsultationInfo.prescriptions.isNotEmpty()) {
+      val medicines = patientTeleconsultationInfo
+          .prescriptions.joinToString(separator = LINE_BREAK) { "${it.name} ${it.dosage}" }
+      message.appendLine("*Current medicines*:")
+          .appendLine(medicines)
+          .appendLine("")
+    }
+  }
+
+  private fun addBloodSugarsSectionToMessage(
+      patientTeleconsultationInfo: PatientTeleconsultationInfoLong,
+      message: StringBuilder
+  ) {
+    if (patientTeleconsultationInfo.bloodSugars.isNotEmpty()) {
+      val bloodSugars = patientTeleconsultationInfo
+          .bloodSugars.joinToString(separator = LINE_BREAK) {
+            val bloodSugarRecordedAtDate = dateFormatter.format(it.recordedAt.toLocalDateAtZone(userClock.zone))
+            val bloodSugarType = resources.getString(it.reading.displayType)
+            val bloodSugarUnit = resources.getString(it.reading.displayUnit)
+
+            "$bloodSugarType ${it.reading.displayValue}${it.reading.displayUnitSeparator}${bloodSugarUnit} ($bloodSugarRecordedAtDate)"
+          }
+
+      val bloodSugarsSize = patientTeleconsultationInfo.bloodSugars.size
+      val bloodSugarsTitle = resources
+          .getQuantityString(R.plurals.patientsummary_contact_doctor_patient_info_blood_sugars, bloodSugarsSize, bloodSugarsSize.toString())
+
+      message.appendLine(bloodSugarsTitle)
+          .appendLine(bloodSugars)
+          .appendLine("")
+    }
+  }
+
+  private fun addBloodPressuresSectionToMessage(
+      patientTeleconsultationInfo: PatientTeleconsultationInfoLong,
+      message: StringBuilder
+  ) {
+    if (patientTeleconsultationInfo.bloodPressures.isNotEmpty()) {
+      val bloodPressures = patientTeleconsultationInfo
+          .bloodPressures.joinToString(separator = LINE_BREAK) {
+            val bpRecordedAtDate = dateFormatter.format(it.recordedAt.toLocalDateAtZone(userClock.zone))
+            "${it.reading.systolic}/${it.reading.diastolic} ($bpRecordedAtDate)"
```

Comment:
Can we extract this to a separate method? The call chain can then be made a lot more readable.

```kotlin
.bloodPressures
.joinToString(LINE_BREAK, ::bloodPressureToDisplayString)
```

Reply:
This applies to the earlier place where we are formatting blood sugars as well.

Reply:
Yes, that's a good idea 👍🏽